UnityProject/Assets/Scripts/Closets/ClosetControl.cs

```diff
@@ -124,31 +153,40 @@ private PushPull PushPull
 	{
 		get
 		{
-			if ( pushPull == null )
+			if (pushPull == null)
 			{
-				Logger.LogErrorFormat( "Closet {0} has no PushPull component! All contained items will appear at HiddenPos!", Category.Transform, gameObject.ExpensiveName() );
+				Logger.LogErrorFormat("Closet {0} has no PushPull component! All contained items will appear at HiddenPos!", Category.Transform, gameObject.ExpensiveName());
 			}
 
 			return pushPull;
 		}
 	}
 
-
 	private void Awake()
 	{
 		EnsureInit();
 		GetComponent<Integrity>().OnWillDestroyServer.AddListener(OnWillDestroyServer);
+
+		if (gameObject.ExpensiveName() != null)
```

Comment:
is this null check needed? ExpensiveName is kinda of expensive method

Reply:
Honestly no idea if there is ever a chance ExpensiveName fetches Null... I saw a few different methods but this seemed cleaner. Couldn't get it to not find the Expensive Name in my testing

Reply:
currently I can't look at the code, but I think ExpensiveName will retrieve the gameobject name in case ObjectAttribute component is not there, which means it will always return something. Can you give it a look at the method definition?

Reply:
Your right, it will get the Initial Name if all else fails, removed it in my latest commit.